src/main/java/frc/robot/systems/AutoHandlerSystem.java

```diff
@@ -0,0 +1,96 @@
+package frc.robot.systems;
+
+public class AutoHandlerSystem {
+	/* ======================== Constants ======================== */
+	// Auto FSM state definitions
+	public enum AutoFSMState {
+		STATE1,
+		STATE2,
+		STATE3
+	}
+
+	/* ======================== Private variables ======================== */
+	//Contains the sequential list of states in the current auto path that must be executed
+	private AutoFSMState[] currentStateList;
+
+	//The index in the currentStateList where the currentState is at
+	private int currentStateIndex;
+
+	//Stores if the current state has finished executing or not
+	private boolean isCurrentStateFinished;
```

Comment:
This doesn't need to be a instance variable, it shouldn't ever need to be used outside update/nextState